cloudbrain/rt_server/rt_server.py

```diff
@@ -0,0 +1,203 @@
+# -*- coding: utf-8 -*-
+
+from cloudbrain.settings import RABBITMQ_ADDRESS
+
+import pika
+import json
+import logging
+
+from sockjs.tornado.conn import SockJSConnection
+from sockjs.tornado import SockJSRouter
+from tornado.ioloop import PeriodicCallback, IOLoop
+from tornado.web import RequestHandler, Application
+
+from cloudbrain.subscribers.SubscriberInterface import Subscriber
+
+SERVER_PORT = 31415
+logging.getLogger().setLevel(logging.INFO)
+
+
+
+class RtStreamConnection(SockJSConnection):
+    """RtStreamConnection connection implementation"""
+    # Class level variable
+    clients = set()
+
+    def __init__(self, session):
+        super(self.__class__, self).__init__(session)
+        self.subscribers = {}
+
+
+    def send_probe_factory(self, metric_name):
+
+        def send_probe(body):
+            logging.debug("GOT: " + body)
+            buffer_content = json.loads(body)
+
+            for record in buffer_content:
+                record["metric"] = metric_name
+                self.send(json.dumps(record))
+
+        return send_probe
+
+
+    def on_open(self, info):
+        logging.info("Got a new connection...")
+        self.clients.add(self)
+        # self.timeout = PeriodicCallback(self.send_heartbeat, 1000)
+        # self.timeout.start()
+
+    # This will receive instructions from the client to change the
+    # stream. After the connection is established we expect to receive a JSON
+    # with deviceName, deviceId, metric; then we subscribe to RabbitMQ and
+    # start streaming the data.
+    #
+    # NOTE: it's not possible to open multiple connections from the same client.
+    #       so in case we need to stream different devices/metrics/etc. at the
+    #       same time, we need to use a solution that is like the multiplexing
+    #       in the sockjs-tornado examples folder.
+    def on_message(self, message):
+        logging.info("Got a new message: " + message)
+
+        stream_configuration = json.loads(message)
```

Comment:
Here we could actually get different type of messages:
* The `stream_configuration` that is the one that is now configured
* The `close_stream` message, that allows us to close a specific subscriber
We can discriminate them using a key (let's say `type`) on the map